data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfiguration.java

```diff
@@ -55,6 +58,12 @@ public class OpenSearchSourceConfiguration {
     @Valid
     private SearchConfiguration searchConfiguration = new SearchConfiguration();
 
+    @JsonProperty("distribution_version")
+    private String distributionVersion;
+
+    @JsonIgnore
+    private DistributionVersion distributionVersionValue;
```

Comment:
We should follow the same pattern we for enums elsewhere to simplify some of this code.

See for example:

https://github.com/opensearch-project/data-prepper/blob/d26748680468aa50bf1bf93c83cc005ae423a2e0/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/configuration/OnErrorOption.java#L14-L34

Maybe add a follow-on PR.